catalog/app/containers/Auth/Layout.tsx

```diff
@@ -0,0 +1,220 @@
+import * as React from 'react'
+import { Link, LinkProps } from 'react-router-dom'
+import * as M from '@material-ui/core'
+import Layout from 'components/Layout'
+import Spinner from 'components/Spinner'
+
+const useContainerStyles = M.makeStyles({
+  root: {
+    marginLeft: 'auto',
+    marginRight: 'auto',
+    maxWidth: 300,
+    width: '100%',
+  },
+})
+
+export function Container(props: JSX.IntrinsicElements['div']) {
+  const classes = useContainerStyles()
+  return <div className={classes.root} {...props} />
+}
+
+export function Heading(props: M.TypographyProps) {
+  return <M.Typography variant="h4" align="center" {...props} />
+}
+
+interface FieldOwnProps {
+  // TODO: use redux-form / final-form type definitions
+  input: {}
+  meta: {
+    error: any
+    submitError: any
```

Comment:
It seems that code expects `error` and `submitError` to be string:

```
    error: string | null
    submitError: string | null
```
or
```
    error?: string
    submitError?: string
```

Reply:
yep, it seems that way in this case, but generally redux-form and final form accepts anything as a validation error